src/main/webapp/resources/script.js

```diff
@@ -12,3 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+$(document).ready(function () {
+  var maxField = 50; //Input fields increment limitation
```

Comment:
maxNumberOfSteps

Actually, define this as a constant: MAX_NUMBER_OF_STEPS outside the document.ready and remove it from the parameters of the functions.